### PR TITLE
MNT: Pybids 0.7 compatibility

### PR DESCRIPTION
**Pull request by effigies**
This branch tracks restoration of functionality to FitLins, using derivatives from fMRIPrep 1.2.1 and models updated according to the draft BIDS model spec.

The ds114 model will be updated as needed in bids-standard/BIDS-examples#130.

Needed modifications to fMRIPrep derivatives are being tracked in poldracklab/fmriprep#1384.

### CODE REVIEW

**Comment by effigies** [comment]
Currently stalled out at bids-standard/pybids#284.

**Comment by adelavega** [comment]
Summary of conversation with Tal:
1) At higher-levels, it looks like to compute contrasts, you feed them into the design matrix as the "intercept" and then computer a one sample t-test on the results. Need to test that this works with a two sample t-test (represented as 1/-1 in BIDS StatsModels)
2) We need to add support for flexible # of higher-levels. Currently only a "fixed" group level is supported. This is necessary for models with more than 1 run per subject (so that estimates are averages for each subject before passing on to group level)
3) We could potentially use the same interface for 1st and higher levels. Need to think about a refactor down the line for this.

**Comment by effigies** [comment]
@adelavega Given that we can now run simple second-level models, should we green-light a pybids 0.7 release?

**Comment by adelavega** [comment]
Ummm... I think the functionality is all there, but are the docs up to date? It's a pretty big API change so that's an important piece

**Comment by adelavega** [comment]
Also, maybe it's time to merge this PR as well, and we can start a new base PR for adding more than 2 levels

**Comment by effigies** [comment]
Okay, let me give it a cleanup review.

**Comment by adelavega** [comment]
The main thing missing from previous functionality is some of the plotting-- design matrix correlation plot (no obvious "EVs"), and contrast matrix plotting.

File: fitlins/interfaces/nistats.py [hunk 1]
```diff
@@ -9,80 +8,45 @@
...
-    contrast_types[init_types.index] = init_types
-
-    if identity:
-        contrast_matrix.loc[identity, identity] = np.eye(len(identity))
-        contrast_types[identity] = 'T'
-
-    return contrast_matrix, contrast_types
+    if not isdefined(contrasts):
+        return []
+    else:
```
Comment (effigies) [inline fitlins/interfaces/nistats.py]: Drop `else` block, as we `return` in the `if` block.

**Comment by effigies** [comment]
There are also some weirdnesses with file naming. e.g.

`derivatives/fitlins/sub-01/ses-retest/sub-01_ses-retest_task-fingerfootlips_bold_contrast-[array([[ 0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ,0. ,  0. ,  1. , -0.5, -0.5,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ,0. ]])]_ortho.png`

File: fitlins/interfaces/nistats.py [hunk 2]
```diff
@@ -190,55 +146,38 @@ class SecondLevelModel(NistatsBaseInterface, SimpleInterface):
...
+        stat_files = _flatten(self.inputs.stat_files)
+        filtered_files = []
+        names = []
+        for m, f in zip(stat_metadata, stat_files):
+            if _match(entities, m):
+                filtered_files.append(f)
+                names.append(m['contrast'])
+
+        for name, weights, type in prepare_contrasts(
+          self.inputs.contrast_info, names):
```
Comment (effigies) [inline fitlins/interfaces/nistats.py]: Move to previous line.

File: requirements.txt [hunk 3]
```diff
@@ -3,7 +3,6 @@ seaborn>=0.7.1
 numpy>=1.11
 nilearn>=0.4.0
 pandas>=0.19
-nipype>=1.1.2
-grabbit>=0.2.3
-pybids>=0.6.5
-git+https://github.com/nistats/nistats.git@ce3695e8f34c6f34323766dc96a60a53b69d2729#egg=nistats
+git+https://github.com/nistats/nistats.git@009ce3fddb3dd01e82bca3ad7d2cdbeece0138f2#egg=nistats
+git+https://github.com/bids-standard/pybids.git@8fd04f3a0443da48cb07357bba0f618fb7f04838#egg=pybids
+git+https://github.com/effigies/nipype.git@5c4e266323a158cf4233eb71f388a0d4b326e51c#egg=nipype
```
Comment (effigies) [inline requirements.txt]: Revert nipype pin.

File: requirements.txt [hunk 4]
```diff
@@ -3,7 +3,6 @@ seaborn>=0.7.1
 numpy>=1.11
 nilearn>=0.4.0
 pandas>=0.19
-nipype>=1.1.2
-grabbit>=0.2.3
-pybids>=0.6.5
-git+https://github.com/nistats/nistats.git@ce3695e8f34c6f34323766dc96a60a53b69d2729#egg=nistats
+git+https://github.com/nistats/nistats.git@009ce3fddb3dd01e82bca3ad7d2cdbeece0138f2#egg=nistats
+git+https://github.com/bids-standard/pybids.git@8fd04f3a0443da48cb07357bba0f618fb7f04838#egg=pybids
```
Comment (effigies) [inline requirements.txt]: Update pybids pin to latest master.

**Comment by adelavega** [comment]
oh, I thought I fixed that. the issue is its using `.format(contrast)` instead of `.format(contrast['name'])`

**Comment by effigies** [comment]
Oh, I think you did. I still had some old outputs in my derivatives directory.

**Pull request merged**